File: ruby solutions/minimum_time_to_build_blocks.rb
# @param {Integer[]} blocks
# @param {Integer} split
# @return {Integer}
def min_build_time(blocks, split)
    while blocks.length > 1
      min1 = blocks.min
      index1 = blocks.index(min1)
      blocks.delete_at(index1)
      
      min2 = blocks.min
      index2 = blocks.index(min2)
      blocks.delete_at(index2)
      
      new_block = [min1, min2].max + split
      blocks.push(new_block)
    end
    blocks[0]
end

File: python solutions/minimum_time_to_build_blocks.py
import heapq
from typing import List

class Solution:
    def minBuildTime(self, blocks: List[int], split: int) -> int:
        heapq.heapify(blocks)
        while len(blocks) > 1:
            x, y = heapq.heappop(blocks), heapq.heappop(blocks)
            heapq.heappush(blocks, max(x, y) + split)

        return blocks[0]